Source/SwiftLintFramework/Rules/Lint/UnusedImportRule.swift

```diff
@@ -195,30 +202,119 @@ private extension File {
                 }
             }
 
-            if let usr = cursorInfo["key.modulename"] as? String {
-                usrFragments.formUnion(usr.split(separator: ".").map(String.init))
-            }
+            appendUsedImports(cursorInfo: cursorInfo, usrFragments: &usrFragments)
         }
+
         // Always disallow 'import Swift' because it's available without importing.
         usrFragments.remove("Swift")
         var unusedImports = imports.subtracting(usrFragments)
         // Certain Swift attributes requires importing Foundation.
         if unusedImports.contains("Foundation") && containsAttributesRequiringFoundation() {
             unusedImports.remove("Foundation")
         }
+
+        if !unusedImports.isEmpty {
+            unusedImports.subtract(
+                operatorImports(
+                    arguments: compilerArguments,
+                    processedTokenOffsets: Set(tokens.map { $0.offset })
+                )
+            )
+        }
+
+        return rangedAndSortedUnusedImports(of: Array(unusedImports), contents: contentsNSString)
+    }
+
+    func rangedAndSortedUnusedImports(of unusedImports: [String], contents: NSString) -> [(String, NSRange)] {
         return unusedImports
             .map { module in
-                let testableImportRange = contentsNSString.range(of: "@testable import \(module)\n")
+                let testableImportRange = contents.range(of: "@testable import \(module)\n")
                 if testableImportRange.location != NSNotFound {
                     return (module, testableImportRange)
                 }
 
-                return (module, contentsNSString.range(of: "import \(module)\n"))
+                return (module, contents.range(of: "import \(module)\n"))
             }
             .sorted(by: { $0.1.location < $1.1.location })
     }
 
-    private func containsAttributesRequiringFoundation() -> Bool {
+    // Operators are omitted in the editor.open request and thus have to be looked up by the indexsource request
+    func operatorImports(arguments: [String], processedTokenOffsets: Set<Int>) -> Set<String> {
```

Comment:
Can you mark these as `private` where possible?

Reply:
Since these methods are a part of a private extension, the compiler warns that using private is redundant in this case. Do you still want me to mark them private?
I could also make the extension non-private and the methods private for the sake of consistency.

Reply:
Oh I didn’t realize these were in a private extension. All good!